### PR TITLE
Feat/azure

### PR DESCRIPTION
**Pull request by MiroslavRepka**
This PR adds Azure provider to a list of supported providers by Claudie.

Please, carefully review the templates, as they are the most important part of this PR.

Also, Azure has a quite small quota on the CPU cores you can request. (I think it's 6 per zone). The testing framework thus supports only one CI run, which is not good. I will open an issue for requesting larger quotas, since this issue is also present on OCI, and possibly on other providers as well.

Edit:
I requested an increase in quotas, which was approved, we should be fine in Azure now.

Edit2:
The tests take much more time than previously (all three test sets ~3h). From my experience, while working on this feature, Azure is much slower than other providers we use (compared to GCP, OCI). We might want to explore creating custom images for VMs, with everything preinstalled (wireguard stuff, node requirement, pre-pulled images for kube-api server, etc). By doing that, we might be able to shave off a lot of time.

### CODE REVIEW

**Review by MarioUhrik — APPROVED** [review]
The templates look quite fine to me :) 
But I didn't check the whole PR thoroughly

**Review by jaskeerat789 — APPROVED** [review]
Changes look fine to me, nice job Miro

File: docs/config/config.md [hunk 1]
```diff
@@ -107,17 +107,21 @@ Node represent a single node from the nodepool
...
-  | UserOcid          | string | OCID of the user                                           |
-  | TenancyOcid       | string | OCID of the tenancy                                        |
-  | OciFingerprint    | string | Fingerprint of the private key saved in `Credentials`      |
-  | CompartmentOcid   | string | OCID of the compartment                                    |
-  | AccessKey         | string | AWS access key                                             |
+  | Name                | Type   | Description                                                        |
+  | ------------------- | ------ | ------------------------------------------------------------------ |
+  | SpecName            | string | Provider name                                                      |
+  | CloudProviderName   | string | Cloud provider name. e.g. `gcp`, `hetzner` or `oci`                |
+  | Credentials         | string | Credentials of the provider                                        |
```
[the remark this answers is not in the document]
Comment (MiroslavRepka) [reply docs/config/config.md:114]: >I think we should mention what value should be passed to this field for each provider explicitly

I agree 

> in the input-manifest.md

not so much here though. Why would we mix `Config` and `Input manifest`? There is a clear distinction for users in Input manifest docs where they should put the credentials for each provider. The Config on the other hand is for developers, who are trying to understand our protobuf spec.

I am all in, to explicitly specify what exactly is in `Credentials`, but let's do it in `config.md`.

**Comment by MarioUhrik** [comment]
Hi guys, why is Azure slow?
Which steps take longer exactly?
Is it because of slower download speeds of tools to install on the machines?

**Comment by MiroslavRepka** [comment]
Hi @MarioUhrik, the Azure is slower in terms of VM creation via Terraform (not only in cluster, on my machine as well) + downloading of the tools takes much longer than in other providers.

**Comment by MarioUhrik** [comment]
I had no idea Azure is much slower than the other providers. That's strange.

Regarding downloading the tools, I might have an idea.
Typically, cloud VMs have a very high download/upload bandwidth.
But it may depend on the type/size of the VM, and other factors.
Perhaps there are other types/sizes of VMs that might behave better.
Still, I am extremely surprised that we are getting any problems with this at all in the first place.
Can you give me an estimate for how much slower bandwidth do the Azure machines have compared to other clouds?

**Comment by MiroslavRepka** [comment]
I did some testing for some of the providers, and here are my findings.

Times to build a simple cluster (2M/2W/no LB), with the VM types the same as in the testing manifest are:
| Provider | Time for testing-framework |
| -------- | -------------------------- |
| GCP      | 732s                       |
| AWS      | 642s                       |
| Hetzner  | 702s                       |
| Azure    | 923s                       |

The biggest difference in Azure was seen in tasks like `Install Wireguard` or `Install kubeadm`, which leads me to the conclusion that network bandwidth is the issue. Also, VMs in Azure took almost 1min to spin up, which is slower than other providers, but not that significant.

We can try to enable something called [`accelerated networking`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_interface#enable_accelerated_networking), we just need to make sure, we will have it enabled only for supported VM types.

**Comment by MarioUhrik** [comment]
That's a nice comparison table :) 

Based on those numbers, it doesn't seem that huge of a difference. 

But Accelerated Networking is free for select few VM types. We should probably use it, in that case.
According to this, the difference can be huge https://www.visualstudiogeeks.com/Azure/WhyUseAcceleratedNwInAzureVms-HowToImproveAzureVmPerf

Maybe we should look into this, either in this issue or the next.

**Pull request merged**